src/domain/app/routes/SilentRenewRedirect.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const SilentRenewRedirect = () => {
+  const navigate = useNavigate();
+  React.useEffect(() => {
+    navigate('/silent_renew.html', { replace: true });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Why is the eslint-disable-next-line necessary? Now the useEffect runs only once, right? But if the `navigate` function was added to the dependency array then the useEffect would be run each time the `navigate` function changes? But when does it change? I tried looking into react-router's useNavigate function (react-router-dom imports useNavigate from react-router according to its source code in node_modules) and `node_modules/react-router/dist/index.js:
```typescript
function useNavigate() {
  let {
    isDataRoute
  } = React.useContext(RouteContext);
  // Conditional usage is OK here because the usage of a data router is static
  // eslint-disable-next-line react-hooks/rules-of-hooks
  return isDataRoute ? useNavigateStable() : useNavigateUnstable();
}
```

So it's a bit, hmm, interesting, let's say. So if `isDataRoute` in `RouteContext` is truish then a stable version of the navigate function is returned:
```typescript
/**
 * Stable version of useNavigate that is used when we are in the context of
 * a RouterProvider.
 */
function useNavigateStable() {
...
```
but otherwise an unstable version.

How do you see this? Did you encounter problems if you added the `navigate` function to the dependency array?

Reply:
I would self use it totally without the useEffect, but a warning was printed to console, that the navigate should always be called from a useEffect-callback. When the dependency array is empty, it really is ran only once, but the redirect should be needed only once. I would say that the way how it's currently used should be fine and it would probably do the same if the the navigate would be in the useEffect, (but I would like to call it only once).